src/AppInstallerCommonCore/Versions.cpp

```diff
@@ -44,25 +82,36 @@ namespace AppInstaller::Utility
                 break;
             }
         }
+
+        THROW_HR_IF(E_INVALIDARG, m_approximateComparator != ApproximateComparator::None && IsBaseVersionUnknown());
     }
 
     bool Version::operator<(const Version& other) const
     {
         // Sort Latest higher than any other values
-        bool thisIsLatest = IsLatest();
-        bool otherIsLatest = other.IsLatest();
+        bool thisIsLatest = IsBaseVersionLatest();
+        bool otherIsLatest = other.IsBaseVersionLatest();
 
-        if (thisIsLatest || otherIsLatest)
+        if (thisIsLatest && otherIsLatest)
         {
-            // If at least one is latest, this can only be less than if the other is and this is not.
+            return ApproximateCompareLessThan(other);
+        }
+        else if (thisIsLatest || otherIsLatest)
+        {
+            // If only one is latest, this can only be less than if the other is and this is not.
             return (otherIsLatest && !thisIsLatest);
         }
 
         // Sort Unknown lower than any known values
-        bool thisIsUnknown = IsUnknown();
-        bool otherIsUnknown = other.IsUnknown();
+        bool thisIsUnknown = IsBaseVersionUnknown();
+        bool otherIsUnknown = other.IsBaseVersionUnknown();

```

Comment:
Is there a reason to change this, since `IsUnknown()` calls `IsBaseVersionUnkown()`

Reply:
In a previous commit, approximate to Unknown is actually supported. I disabled it after internal discussion about what we would do to Unknown. I kept it for parity. And in case we need it in the future.